src/js/modules/infragistics.ui.editors.js

```diff
@@ -4460,22 +4461,31 @@
 				this.options.minDecimals;
 		},
 		_applyOptions: function () { // NumericEditor
-			var delta, fractional, initialValue;
+
 			this._super();
-			initialValue = this.options.value;
+			this._validateSpinSettings();
+			this._validateDecimalSettings();
+
+			if (this.options.maxLength !== null) {
+				this.options.maxLength = null;
+			}
+			if (this.options.value < 0) {
+				this._editorInput.addClass(this.css.negative);
+			}
+		},
+		_validateSpinSettings: function() {
+			var delta, fractional;
 
 			// A.M. October 11 2016 #420 "Spin button increase/decrease button not disabled"
 			if (this.options.buttonType === "spin") {
-				this._setSpinButtonsState(initialValue);
+				this._setSpinButtonsState(this.options.value);
 			}
 			if (this.options.spinDelta !== 1) {
 				delta = this.options.spinDelta;
 				if (typeof delta !== "number") {
-					this.options.spinDelta = 1;
```

Comment:
These should probably stay - at this stage the Error won't break the editor entirely, so spin still needs a delta 😕
(whether that should be the expected behavior is a different issue)